server/bleep/src/semantic.rs

```diff
@@ -420,73 +457,66 @@ impl Semantic {
         );
         debug!(chunk_count = chunks.len(), "found chunks");
 
-        // Prepend all chunks with `repo_name   relative_path`
-        let chunk_prefix = format!("{repo_name}\t{relative_path}\n");
-
-        let datapoints = chunks
-            .par_iter()
-            .filter_map(
-                |chunk| match self.embed(&(chunk_prefix.clone() + chunk.data)) {
-                    Ok(ok) => Some(PointStruct {
-                        id: Some(PointId::from(uuid::Uuid::new_v4().to_string())),
-                        vectors: Some(ok.into()),
-                        payload: Payload {
-                            lang: lang_str.to_ascii_lowercase().into(),
-                            repo_name: repo_name.into(),
-                            repo_ref: repo_ref.into(),
-                            relative_path: relative_path.into(),
-                            branches: branches.to_owned(),
-                            text: chunk.data.into(),
-                            start_line: chunk.range.start.line as u64,
-                            end_line: chunk.range.end.line as u64,
-                            start_byte: chunk.range.start.byte as u64,
-                            end_byte: chunk.range.end.byte as u64,
-                            ..Default::default()
-                        }
-                        .into_qdrant(),
-                    }),
-                    Err(err) => {
-                        warn!(?err, %chunk_prefix, "embedding failed");
-                        None
-                    }
-                },
-            )
-            .collect::<Vec<_>>();
+        let embedder = |c: &str| {
+            info!("generating embedding");
+            self.embed(c)
+        };
+        chunks.par_iter().for_each(|chunk| {
+            let data = format!(
+                "{repo_name}\t{relative_path}\n{unique_hash}\n{}",
```

Comment:
Why is `unique_hash` here? This could negatively impact retrieval.

Reply:
It's being used to pin the chunk to the containing file. I can't reason either way about retrieval performance (although I think this is in the *probably fine* territory), but I guess we could construct the reference hash differently, and achieve the same effect.

Reply:
We don't want to embed it. `repo_name` and `relative_path` are included in `data` to improve retrieval (by matching on these fields).